pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -10197,3 +10197,27 @@ func TestChangefeedProtectedTimestampUpdate(t *testing.T) {
 
 	cdcTest(t, testFn, feedTestForceSink("kafka"), withTxnRetries)
 }
+
+// TestChangefeedDiffEmptyTable verifies that a changefeed with the diff option
+// correctly captures changes on an initially empty table. It ensures changefeed
+// behavior is consistent when tables start with no data.
+func TestChangefeedDiffEmptyTable(t *testing.T) {
```

Comment:
this test tests that `create changefeed for empty` works, but this already works. the issue is with `create changefeed as select * from empty`. can you add a test for this?

also, i think the correct behaviour would either be 1) error more gracefully (not an "unexpected error" -- maybe an unimplemented error?), or 2) succeed and have rows look like this: `{"key":"[1053964827675656193]","value":"{\"after\": {}"}` (with after being empty obj, or null, or similar). 

can you implement one of the above and test it?

Reply:
Fixed